### PR TITLE
Update applicant screens

### PR DESCRIPTION
**Pull request by Davidslv**
Story: https://mojdigital.teamwork.com/#/tasks/20230244

Screenshot 2020-04-23 at 18 12 17
Screenshot 2020-04-23 at 18 12 48
Screenshot 2020-04-23 at 18 13 02
Screenshot 2020-04-23 at 18 13 49

### CODE REVIEW

File: app/views/steps/applicant/contact_details/edit.html.erb [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
 
-    <h1 class="heading-xlarge gv-u-heading-xxlarge"><%=t '.heading', name: @form_object.record.full_name %></h1>
+    <h1 class="govuk-heading-xl"><%=t '.heading', name: @form_object.record.full_name %></h1>
 
-    <%= step_form @form_object do |f| %>
-      <%= f.text_field :email, class: 'narrow' %>
-      <%= f.text_field :home_phone %>
-      <%= f.text_field :mobile_phone %>
+    <%= step_form @form_object, builder: GOVUKDesignSystemFormBuilder::FormBuilder do |f| %>
+      <%= f.govuk_email_field :email, label: { size: 's' } %>
```
Comment (zheileman) [inline app/views/steps/applicant/contact_details/edit.html.erb:11]: Same as above for labels size an for the legend in the radios.

File: app/views/steps/applicant/contact_details/edit.html.erb [hunk 2]
```diff
@@ -1,17 +1,21 @@
...
+      <%= f.govuk_text_field :home_phone, label: { size: 's' } %>
```
Comment (zheileman) [inline app/views/steps/applicant/contact_details/edit.html.erb:12]: Let's use for phone fields, when possible, `govuk_phone_field`:
https://govuk-form-builder.netlify.app/form-elements/text-field/

File: app/views/steps/applicant/contact_details/edit.html.erb [hunk 3]
```diff
@@ -1,17 +1,21 @@
...
+      <%= f.govuk_text_field :mobile_phone, label: { size: 's' } %>
 
-      <%= f.radio_button_fieldset :voicemail_consent, choices: GenericYesNo.values %>
+      <%= f.govuk_radio_buttons_fieldset :voicemail_consent, legend: { tag: 'span', size: 's' } do %>
```
Comment (zheileman) [inline app/views/steps/applicant/contact_details/edit.html.erb:15]: When the radios do not reveal anything (it is just a list of radios, 2 or more) you can make it shorter with this other form, like we do in other places like app/views/steps/screener/parent/edit.html.erb:

```ruby
<%= f.govuk_collection_radio_buttons :voicemail_consent, GenericYesNo.values, :value, nil %>
```

File: app/views/steps/applicant/has_solicitor/edit.html.erb [hunk 4]
```diff
@@ -1,18 +1,21 @@
...
-    <h1 class="heading-xlarge gv-u-heading-xxlarge">
-      <%= heading %>
-    </h1>
+    <h1 class="govuk-heading-xl"><%= heading %></h1>
 
-    <p class="govuk-govspeak gv-s-prose"><%=t '.lead_text' %></p>
+    <p class="govuk-body"><%=t '.lead_text' %></p>
+
+    <%= step_form @form_object, builder: GOVUKDesignSystemFormBuilder::FormBuilder do |f| %>
+      <%= f.govuk_radio_buttons_fieldset :has_solicitor, legend: { tag: 'span', size: 's' } do %>
```
Comment (zheileman) [inline app/views/steps/applicant/has_solicitor/edit.html.erb:14]: Same as above, we can use the shorter form `govuk_collection_radio_buttons` here.

And can we incorporate the heading and `lead_text` into the fieldset, like we do for example here:
steps/abuse_concerns/question/edit.html.erb

File: app/views/steps/applicant/personal_details/edit.html.erb [hunk 5]
```diff
@@ -1,27 +1,30 @@
...
-    <%= step_form @form_object do |f| %>
-      <%=
-        f.radio_button_fieldset :has_previous_name, inline: true do |fieldset|
-          fieldset.radio_input GenericYesNo::YES, panel_id: :previous_name_panel
-          fieldset.radio_input GenericYesNo::NO
-          fieldset.revealing_panel(:previous_name_panel) { |panel| panel.text_field :previous_name }
-        end
-      %>
+    <%= step_form @form_object, builder: GOVUKDesignSystemFormBuilder::FormBuilder do |f| %>
+      <%= f.govuk_radio_buttons_fieldset :has_previous_name, legend: { tag: 'span', size: 's'  } do %>
```
Comment (zheileman) [inline app/views/steps/applicant/personal_details/edit.html.erb:11]: Same as above for the size of legends and labels 😄

File: app/views/steps/applicant/personal_details/edit.html.erb [hunk 6]
```diff
@@ -1,27 +1,30 @@
...
+        <%= f.govuk_radio_button :has_previous_name, GenericYesNo::YES, link_errors: true do %>
+          <%= f.govuk_text_field :previous_name, label: { size: 's' } %>
+        <% end %>
+        <%= f.govuk_radio_button :has_previous_name, GenericYesNo::NO %>
+      <% end %>
 
-      <%= f.radio_button_fieldset :gender, choices: Gender.values %>
-      <%= f.gov_uk_date_field :dob, placeholders: true, legend_text: t('shared.form_elements.dob'), form_hint_text: t('shared.form_elements.dob_applicant_hint') %>
-      <%= f.text_field :birthplace %>
+      <%= f.govuk_radio_buttons_fieldset :gender, legend: { tag: 'span', size: 's' } do %>
```
Comment (zheileman) [inline app/views/steps/applicant/personal_details/edit.html.erb:18]: Like above, collections without anything revealing, better this way:
https://github.com/ministryofjustice/c100-application/blob/develop/app/views/steps/children/personal_details/edit.html.erb#L19

File: app/views/steps/applicant/personal_details/edit.html.erb [hunk 7]
```diff
@@ -1,27 +1,30 @@
...
+        <%= f.govuk_radio_button :gender, Gender::FEMALE, link_errors: true %>
+        <%= f.govuk_radio_button :gender, Gender::MALE %>
+        <%= f.govuk_radio_button :gender, Gender::UNSPECIFIED %>
+      <% end %>
 
-      <div class="xform-group form-submit">
-        <%= f.continue_button %>
-      </div>
+      <%= f.govuk_date_field :dob, hint_text: t('shared.form_elements.order_issue_hint'), legend: { size: 's', text: t('shared.form_elements.dob') } %>
```
Comment (zheileman) [inline app/views/steps/applicant/personal_details/edit.html.erb:24]: I've been removing the parametrised hint and legend text in favour of using the locales. Take a look at any other date already migrated.

File: config/locales/en.yml [hunk 8]
```diff
@@ -27,6 +27,11 @@ en:
       'no': 'No'
       unknown: *dont_know
 
+    GENDER: &GENDER
```
Comment (zheileman) [inline config/locales/en.yml:30]: No need for this, the gender locales are already part of the `PERSONAL_DETAILS_FIELDS` dictionary.

File: config/locales/en.yml [hunk 9]
```diff
@@ -1287,14 +1281,23 @@ en:
       steps_applicant_personal_details_form:
         <<: *PERSONAL_DETAILS_FIELDS
         previous_name: Enter your previous name
+        has_previous_name_options:
```
Comment (zheileman) [inline config/locales/en.yml:1284]: Better if we do the same as we did with gender, leave it in the dictionary but for now have an old and a new version so it continues working on the forms not yet migrated.

File: config/locales/en.yml [hunk 10]
```diff
@@ -103,8 +108,6 @@ en:
       country: Country
       postcode: Postcode
       address_unknown: I don’t know where they currently live
-      residence_requirement_met:
```
Comment (zheileman) [inline config/locales/en.yml:106]: Same here, if we can have an old version (the removed one) and a new version (adding _options) like we have in `PERSONAL_DETAILS_FIELDS` then it is shared between all the people (applicant, respondent, other parties...) as this is used in multiple places.

**Comment by zheileman** [comment]
Another thing, without the changes to the form object, the new Dates will not work.

**Comment by Davidslv** [comment]
@zheileman I've address all your comments, thank you!

File: app/views/steps/applicant/address_details/edit.html.erb [hunk 11]
```diff
@@ -1,24 +1,25 @@
...
-    <%= step_form @form_object do |f| %>
-      <%= f.text_field :address_line_1 %>
-      <%= f.text_field :address_line_2, label_options: { class: 'visually-hidden' } %>
-      <%= f.text_field :town %>
-      <%= f.text_field :country %>
-      <%= f.text_field :postcode %>
+    <%= step_form @form_object, builder: GOVUKDesignSystemFormBuilder::FormBuilder do |f| %>
+      <%= f.govuk_text_field :address_line_1 %>
+      <%= f.govuk_text_field :address_line_2, label: { hidden: true } %>
+      <%= f.govuk_text_field :town %>
```
Comment (zheileman) [inline app/views/steps/applicant/address_details/edit.html.erb:13]: Forgot to mention before, but we will need to set the witdth of town, country and postcode, as per example here:
https://design-system.service.gov.uk/patterns/addresses/multiple/index.html

I reckon for postcodes we've been using `width: 'one-third'` so for country and town we can use `width: 'one-half'` maybe? Give it a try and see how it feels.

File: app/views/steps/applicant/contact_details/edit.html.erb [hunk 12]
```diff
@@ -1,17 +1,18 @@
...
-    <h1 class="heading-xlarge gv-u-heading-xxlarge"><%=t '.heading', name: @form_object.record.full_name %></h1>
+    <h1 class="govuk-heading-xl"><%=t '.heading', name: @form_object.record.full_name %></h1>
 
-    <%= step_form @form_object do |f| %>
-      <%= f.text_field :email, class: 'narrow' %>
-      <%= f.text_field :home_phone %>
-      <%= f.text_field :mobile_phone %>
+    <%= step_form @form_object, builder: GOVUKDesignSystemFormBuilder::FormBuilder do |f| %>
+      <%= f.govuk_email_field :email %>
+      <%= f.govuk_phone_field :home_phone %>
```
Comment (zheileman) [inline app/views/steps/applicant/contact_details/edit.html.erb:12]: Same here for the width, we can use as example `steps/solicitor/contact_details/edit.html.erb`

**Pull request merged**